`--follow-imports=silent` is not silent if imported file is in cache.

**Bug Report**

When a file has been previously analyzed, and errors were found, and that result was cached,
(A clear and concise description of what the bug is.)

**To Reproduce**

Similar to https://github.com/python/mypy/blob/704aee63c54cceb00c4c60699dc5afa493899d50/test-data/unit/check-incremental.test#L2451, but with an extra `cmd3`/`out3` equal to the first.

Create two files:

```
# a.py
import b

b.foo(1, 2)
```

and

```
# b.py
def foo(a: int, b: int) -> str:
    return a + b
```

Then, run the following three commands (in this order):

1. `mypy --follow-imports=silent a.py`
2. `mypy --follow-imports=silent b.py`
3. `mypy --follow-imports=silent a.py` (same as first command)

(A more 'minimal' example would be to not run `mypy --follow-imports=silent a.py` first, which still exhibits the bug, but I think running 'a', 'b', 'a' more clearly shows the bug).

*Expected Behavior**

Only run 2 (the one directly analyzing `b.py`) produces output.

```
# mypy --follow-imports=silent a.py
Success: no issues found in 1 source file
# mypy --follow-imports=silent b.py
b.py:2: error: Incompatible return value type (got "int", expected "str")  [return-value]
Found 1 error in 1 file (checked 1 source file)
# mypy --follow-imports=silent a.py
Success: no issues found in 1 source file
```

**Actual Behavior**

```
# mypy --follow-imports=silent a.py
Success: no issues found in 1 source file
# mypy --follow-imports=silent b.py
b.py:2: error: Incompatible return value type (got "int", expected "str")  [return-value]
Found 1 error in 1 file (checked 1 source file)
# mypy --follow-imports=silent a.py
b.py:2: error: Incompatible return value type (got "int", expected "str")  [return-value]
Found 1 error in 1 file (checked 1 source file)
```

Note how the third run (which uses `--follow-imports=silent`) is *not* silent about errors in not explicitly listed files, while the first was.

**Your Environment**

<!-- Include as many relevant details about the environment you experienced the bug in -->

- Mypy version used: 1.19.1 (but reproducible on master)
- Mypy command-line flags: `--follow-imports=silent`
- Mypy configuration options from `mypy.ini` (and other config files): None.
- Python version used: 3.13.11

--- a/mypy/build.py
+++ b/mypy/build.py
@@ -2180,7 +2180,8 @@ class State:
             self.dep_hashes = {
                 k: v for (k, v) in zip(self.meta.dependencies, self.meta.dep_hashes)
             }
-            self.error_lines = self.meta.error_lines
+            # Only copy `error_lines` if the module is not silently imported.
+            self.error_lines = [] if self.ignore_all else self.meta.error_lines
             if temporary:
                 self.load_tree(temporary=True)
             if not manager.use_fine_grained_cache():

--- a/test-data/unit/check-incremental.test
+++ b/test-data/unit/check-incremental.test
@@ -2465,6 +2465,25 @@ def foo(a: int, b: int) -> str:
 [out2]
 tmp/b.py:2: error: Incompatible return value type (got "int", expected "str")
 
+[case testIncrementalWithSilentImportsReverse]
+-- Same commands as previous test, just in reverse order.
+-- expected output is also in reverse order
+# cmd: mypy -m b
+# cmd2: mypy -m a
+# flags: --follow-imports=silent
+[file a.py]
+import b
+
+b.foo(1, 2)
+
+[file b.py]
+def foo(a: int, b: int) -> str:
+    return a + b
+
+[out]
+tmp/b.py:2: error: Incompatible return value type (got "int", expected "str")
+[out2]
+
 [case testForwardNamedTupleToUnionWithOtherNamedTUple]
 from typing import NamedTuple, Union
 

